src/feat/wave-reader.cc

```diff
@@ -221,55 +221,88 @@ void WaveData::Read(std::istream &is) {
+                  << " + " << data_chunk_size << " bytes "
+                  << "(we do not support reading multiple data chunks).";
   }
 
-  std::vector<char*> data_pointer_vec;
-  std::vector<int> data_size_vec;
+  std::vector<char> chunk_data_vec;
   uint32 num_bytes_read = 0;
-  for (int32 remain_chunk_size = data_chunk_size; remain_chunk_size > 0;
-       remain_chunk_size -= kBlockSize) {
-    int32 this_block_size = remain_chunk_size;
-    if (kBlockSize < remain_chunk_size)
-      this_block_size = kBlockSize;
-    char *block_data_vec = new char[this_block_size];
-    is.read(block_data_vec, this_block_size);
-    num_bytes_read += is.gcount();
-    data_size_vec.push_back(is.gcount());
-    data_pointer_vec.push_back(block_data_vec);
-    if (num_bytes_read < this_block_size)
-      break;
-  }
-
-  std::vector<char> chunk_data_vec(num_bytes_read);
-  uint32 data_address = 0;
-  for (int i = 0; i < data_pointer_vec.size(); i++) {
-    memcpy(&(chunk_data_vec[data_address]), data_pointer_vec[i],
-           data_size_vec[i]);
-    delete[] data_pointer_vec[i];
-    data_address += data_size_vec[i];
+  if (!is_stream_mode) {
+     std::vector<char*> data_pointer_vec;
+     std::vector<int> data_size_vec;
+     num_bytes_read = 0;
+     for (int32 remain_chunk_size = data_chunk_size; remain_chunk_size > 0;
+          remain_chunk_size -= kBlockSize) {
+       int32 this_block_size = remain_chunk_size;
+       if (kBlockSize < remain_chunk_size)
+         this_block_size = kBlockSize;
+       char *block_data_vec = new char[this_block_size];
+       is.read(block_data_vec, this_block_size);
+       num_bytes_read += is.gcount();
+       data_size_vec.push_back(is.gcount());
+       data_pointer_vec.push_back(block_data_vec);
+       if (num_bytes_read < this_block_size)
+         break;
+     }
+     chunk_data_vec.resize(num_bytes_read);
+     uint32 data_address = 0;
+     for (int i = 0; i < data_pointer_vec.size(); i++) {
+       KALDI_ASSERT(data_address + data_size_vec[i] <= chunk_data_vec.size());
+       memcpy(&(chunk_data_vec[data_address]), data_pointer_vec[i], data_size_vec[i]);
+       delete[] data_pointer_vec[i];
+       data_address += data_size_vec[i];
+     }
+
+     if (num_bytes_read == 0 && num_bytes_read != data_chunk_size) {
+       KALDI_ERR << "WaveData: failed to read data chunk (read no bytes)";
+     } else if (num_bytes_read != data_chunk_size) {
+       KALDI_ASSERT(num_bytes_read < data_chunk_size);
+       KALDI_WARN << "Read fewer bytes than specified in the header: "
+                  << num_bytes_read << " < " << data_chunk_size;
+     }
+  } else {
+    // when data are read in stream model, we don't know size in advance
+    // data_chunk_size and riff_chunk_size (Uint32) could be 
+    // either ox00000000 or oxFFFFFFFF
+    // we need to read in the stream differently in this case
+    chunk_data_vec.clear();
+    num_bytes_read = 0;
+    while (is.good()) {
+      char buffer[256];
```

Comment:
I think you shouild better use s/256/kBlockSize/ -- the constant is already defined.